### DISCUSSION TITLE
Execute two commands with one binding?

### DISCUSSION THREAD

**Discussion by txtyash** [gdh1995/vimium-c#1102] [Q&A] [answered]
When I press Alt+r I want to escape and then I want to switch to the previous tab. This doesn`t work:
```
map <a-r> reset visitPreviousTab
```

If I just simply add
```
map <a-r> visitPreviousTab
```
then this doesn't work when the focus is in a textarea or any input

**Comment by shynekomaid** [comment]
So. In the new version of chrome, the "Save tab group" feature has broken the tab switching mechanism, this could be your problem. Also try using ^ to switch tabs.

**Comment by shynekomaid** [comment]
For chain binding, read - https://github.com/gdh1995/vimium-c/wiki/Auto-run-a-tree-of-commands

**Reply by txtyash** [reply]
I couldn't understand everything but I came up with this:
```
mapKey <a-r> <esc> $then="visitPreviousTab"
```
which didn't really work

**Comment by gdh1995** [comment, marked as answer]
To jump across tabs whatever mode Vimium C is in (normal, insert, or anything else), the recommended way is to use `chrome://extensions/shortcuts` and `shortcut userCustomized1 visitPreviousTab`

If a key mapping uses `<a-r>`, then it only takes effects in normal mode, but not in insert mode.
To trigger commands in insert mode, you'll need `:i` to indicate it's insert-mode-enabled:
```
# both lines are necessary, unless you use `shortcut userCustomized1 ...`
map <a-r> visitPreviousTab
map <a-r:i> visitPreviousTab

# or you may use the new syntax of `map!`
map! <a-r> visitPreviousTab
```

`$then` is a parameter of `map` (and `run`), but not `mapKey`; `<esc>` is a key but not a command name. So, `map <a-r:i> dispatchEvent esc $then="visitPreviousTab"` may help.
